clients/gateway.go

```diff
@@ -42,7 +83,21 @@ func (c *GatewayClient) buildQueryString(endpoint string, args map[string]string
 
 // get performs a "GET" http request with the given URL and returns the response body
 func (c *GatewayClient) get(queryUrl string) ([]byte, error) {
-	res, err := http.Get(queryUrl)
+	var res *http.Response
+	var err error
+	wait := c.minWait
+	for {
```

Comment:
Instead of breaking on `maxWait`, I think we should put a limit on number of retries

Reply:
Do you mean replacing `maxWait` with something like `maxRetries`? Just want to make sure I'm understanding.